### PR TITLE
Relocate IP Lookup to ensure it is always set

### PR DESCRIPTION
**Pull request by stefanbenten**
What: Move IP Lookup to the cache Put Function

Why: Ensure the IP is always up to date and even set!

Thanks for submitting a PR!

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

File: pkg/transport/transport.go [hunk 1]
```diff
@@ -105,12 +105,6 @@ func (transport *Transport) DialNode(ctx context.Context, node *pb.Node, opts ..
 		return nil, Error.Wrap(err)
 	}
 
-	ipAddr, err := getIP(conn.Target())
-	if err != nil {
-		return nil, err
-	}
-	node.LastIp = ipAddr
```
Comment (ifraixedes) [inline pkg/transport/transport.go]: I'm not sure if this part should be removed, I understand that the IP should be updated when the node is saved in the cache, however shouldn't we set the such filed when the node is dialed successful?

Comment (stefanbenten) [reply pkg/transport/transport.go]: @ifraixedes The IP is being filled further down the code in the alertSuccessful Method, as it is calling the cache PUT function. So it is called still when dialing the node.

**Comment by stefanbenten** [comment]
Testplanet doesnt allow the configuration of other hostnames for the nodes, so the tests are now failing. This is blocking the successful build

File: pkg/transport/transport.go (hunk 1 above)
Comment (ifraixedes) [reply pkg/transport/transport.go]: always? or it depends of the observers?

Comment (stefanbenten) [reply pkg/transport/transport.go]: We were depending on the observers before already.

File: pkg/overlay/selection_test.go [hunk 2]
```diff
@@ -221,17 +219,15 @@ func TestNodeSelection(t *testing.T) {
 }
 
 func TestDistinctIPs(t *testing.T) {
+	if runtime.GOOS == "darwin" {
```
Comment (stefanbenten) [inline pkg/overlay/selection_test.go]: I have to check that on Windows, but that might not work there as well :)

File: internal/testplanet/reconfigure.go [hunk 3]
```diff
@@ -21,6 +21,7 @@ type Reconfigure struct {
 
 	NewStorageNodeDB func(index int) (storagenode.DB, error)
 	StorageNode      func(index int, config *storagenode.Config)
+	NewIPCount       int
```
Comment (stefanbenten) [inline internal/testplanet/reconfigure.go]: Nice!

**Pull request merged**